main.tf

```diff
@@ -1,12 +1,15 @@
 locals {
   tfe_workspace = {
     clear_text_terraform_variables = var.account.environment != null ? {
-      account = var.name, environment = var.account.environment } : {
-      account = var.name
+      account               = var.name,
+      environment           = var.account.environment,
+      workload_boundary_arn = try(aws_iam_policy.workload_boundary[0].arn, null)
```

Comment:
hey hey if you don't provide a permission boundary this will result in an empty variable called `workload_boundary_arn` to the workspace, while I think what you are trying to achieve is no variable added to the workspace when you don't specify a permissions boundary. Thats why we also added this if structure for var.account.environment. So that needs to be solved, or we need to go for a data resource in the accounts. 

The if/if else/else structure is always a bit hard in terraform, let me check with stephen tomorrow how to solve this